File: Twitter/Data/ProfileInfo.cs
using Core.Crawling;
using Core.Data;
using System;

namespace Twitter.Data
{
    public class ProfileInfo : IProfileInfo
    {
        public string Social => "twitter";
        public string Link { get; set; }

        public string Id;
        public string Name;
        public string Description;
        public string Location;
        public string Url;
        public string JoinDate => DateUtils.TryParseDate(RawJoinDate);

        public ImageUrl HeaderImg;
        public ImageUrl PhotoImg;

        public string RawFollowing;
        public string RawFollowers;
        public string RawJoinDate;

        public int Following => NumberUtils.ParseCount(RawFollowing);
        public int Followers => NumberUtils.ParseCount(RawFollowers);

        public DateTimeOffset CreatedAt = DateTimeOffset.UtcNow;

        public static ProfileInfo Collect(Browser browser)
        {
            return browser.RunCollector<ProfileInfo>("Scripts/Twitter/ProfileInfo.js");
        }
    }
}

File: Twitter/Scripts/Twitter/ProfileInfo.js
﻿
var username = document.querySelector('[data-testid="UserName"]');
if (!username) {
    return "null";
}

var description = document.querySelector('[data-testid="UserDescription"]');
var location = document.querySelector('[data-testid="UserLocation"]');
var url = document.querySelector('[data-testid="UserUrl"]');
var joindate = document.querySelector('[data-testid="UserJoinDate"]');

var header = document.querySelector('[href="' + document.location.pathname + '/header_photo"] img');
var photo = document.querySelector('[href="' + document.location.pathname + '/photo"] img');
var following = document.querySelector('[href="' + document.location.pathname + '/following"]');
var followers = document.querySelector('[href="' + document.location.pathname + '/followers"]');

var { 0: name, 1: id } = username?.innerText?.split('\n');

// Unwrap emoji
description?.querySelectorAll('img')?.forEach(function (img) {
    img.outerHTML = img.alt
});

// Unwrap links
description?.querySelectorAll('a')?.forEach(function (a) {
    if (a.href.startsWith('https://t.co/')) {
        a.outerHTML = a.href
    }
});

return JSON.stringify({
    Link: document.location.href,

    Id: id?.trim(),
    Name: name?.trim(),
    Description: description?.innerText?.trim(),
    Location: location?.innerText?.trim(),
    Url: url?.href,

    HeaderImg: header?.src,
    PhotoImg: photo?.src,

    RawFollowing: following?.querySelector('span')?.innerText,
    RawFollowers: followers?.querySelector('span')?.innerText,
    RawJoinDate: joindate?.innerText?.trim(),
})